### PR TITLE
Refactor chart test

### PR DESCRIPTION
**Pull request by gliptak**
Signed-off-by: Gábor Lipták <gliptak@gmail.com>

**If applicable**:
- [x] this PR contains unit tests

### CODE REVIEW

**Comment by hickeyma** [comment]
Tests passed as expected. LGTM.

**Review by michelleN — COMMENTED** [review]
@gliptak I'm not sure why the circle build wasn't run. Would you mind force-pushing you branch again to see if that fixes the issue? maybe the webhook just didn't fire off.

**Review by hickeyma — COMMENTED** [review]
IIRC, close/reopen the PR should kick off a build.

**Comment by michelleN** [comment]
@hickeyma I tried closing and then re-opening it and it didn't look like that worked unfortunately.

**Comment by gliptak** [comment]
@michelleN might have to do something with having CircleCI configured against my fork at ```gliptak/helm```.

Look at https://github.com/helm/helm/pull/5289 It triggers CircleCI build on ```gliptak/helm``` ... I think it is a CircleCI bug. I did open a ticket with them.

**Comment by gliptak** [comment]
I deconfigured CircleCI for my fork, now the build passes.

File: pkg/chartutil/create_test.go [hunk 1]
```diff
@@ -26,6 +26,14 @@ import (
 	"k8s.io/helm/pkg/proto/hapi/chart"
 )
 
+func checkFile(t *testing.T, dir string, f string) {
```
Comment (bacongobbler) [inline pkg/chartutil/create_test.go]: can we refactor this into

```
func checkFile(t *testing.T, f string) {
```

Then change every function call to the full filepath? e.g.

```
checkFile(t, filepath.Join(dir, f))
```

There's no need to separate `dir` and `f` if we're just going to call `filepath.Join` a second time.

File: pkg/chartutil/create_test.go [hunk 2]
```diff
@@ -26,6 +26,14 @@ import (
...
+	if fi, err := os.Stat(filepath.Join(dir, f)); err != nil {
+		t.Errorf("Expected %s file: %s", f, err)
```
Comment (bacongobbler) [inline pkg/chartutil/create_test.go]: With the other comment in place, this as well as line 33 can be changed to use `filepath.Base(f)` to get the file name.

**Review by bacongobbler — CHANGES_REQUESTED** [review]

File: pkg/chartutil/create_test.go (hunk 1 above)
Comment (gliptak) [reply pkg/chartutil/create_test.go]: updated

**Review by hickeyma — CHANGES_REQUESTED** [review]
Sorry but missed this previously. I wonder could you also update `TestCreateFrom` to call checkFile()? Then, it might make sense to also add a CheckDir function.

**Comment by gliptak** [comment]
@hickeyma thank you for the pointers. i made the updates

**Review by hickeyma — APPROVED** [review]
Thanks. LGTM.

**Comment by gliptak** [comment]
@bacongobbler would you like to see additional changes?

**Pull request closed without merging**